homeassistant/components/cover/template.py

```diff
@@ -0,0 +1,353 @@
+                template_entity_ids |= set(temp_ids)
+
+        if tilt_template is not None:
+            temp_ids = tilt_template.extract_entities()
+            if str(temp_ids) != MATCH_ALL:
+                template_entity_ids |= set(temp_ids)
+
+        if icon_template is not None:
+            temp_ids = icon_template.extract_entities()
+            if str(temp_ids) != MATCH_ALL:
+                template_entity_ids |= set(temp_ids)
+
+        if not template_entity_ids:
+            template_entity_ids = MATCH_ALL
+
+        entity_ids = device_config.get(CONF_ENTITY_ID, template_entity_ids)
+
+        covers.append(
+            CoverTemplate(
+                hass,
+                device, friendly_name, state_template,
+                position_template, tilt_template, icon_template,
+                open_action, close_action, stop_action,
+                position_action, tilt_action, entity_ids
+            )
+        )
+    if not covers:
+        _LOGGER.error("No covers added")
+        return False
+
+    async_add_devices(covers, True)
+    return True
+
+
+class CoverTemplate(CoverDevice):
+    """Representation of a Template cover."""
+
+    def __init__(self, hass, device_id, friendly_name, state_template,
+                 position_template, tilt_template, icon_template,
+                 open_action, close_action, stop_action,
+                 position_action, tilt_action, entity_ids):
+        """Initialize the Template cover."""
+        self.hass = hass
+        self.entity_id = async_generate_entity_id(
+            ENTITY_ID_FORMAT, device_id, hass=hass)
+        self._name = friendly_name
+        self._template = state_template
+        self._position_template = position_template
+        self._tilt_template = tilt_template
+        self._icon_template = icon_template
+        self._open_script = Script(hass, open_action)
+        self._close_script = Script(hass, close_action)
+        self._stop_script = Script(hass, stop_action)
+        self._position_script = None
+        if position_action is not None:
+            self._position_script = Script(hass, position_action)
+        self._tilt_script = None
+        if tilt_action is not None:
+            self._tilt_script = Script(hass, tilt_action)
+        self._icon = None
+        self._position = None
+        self._tilt_value = None
+        self._entities = entity_ids
+
+        if self._template is not None:
+            self._template.hass = self.hass
+        if self._position_template is not None:
+            self._position_template.hass = self.hass
+        if self._tilt_template is not None:
+            self._tilt_template.hass = self.hass
+        if self._icon_template is not None:
+            self._icon_template.hass = self.hass
+
+    @asyncio.coroutine
+    def async_added_to_hass(self):
+        """Register callbacks."""
+        state = yield from async_get_last_state(self.hass, self.entity_id)
+        if state:
```

Comment:
Sorry, added comment to the wrong line. But `state.state` is going to be `STATE_OPEN` or `STATE_CLOSED`. Both are string values and thus will always be truthy, setting the position to 100.